prime-angular-frontend/src/app/modules/paper-enrolment/shared/guards/paper-enrolment.guard.ts

```diff
@@ -26,29 +27,39 @@ export class PaperEnrolmentGuard extends BaseGuard {
   }
 
   protected checkAccess(routePath: string = null, params: Params): Observable<boolean> | Promise<boolean> {
-    const enrolleeId = params.eid;
-    if (!enrolleeId) {
+    const enrolleeId = +params.eid;
+    if (enrolleeId) {
       return this.paperEnrolmentResource.getEnrolleeById(enrolleeId)
         .pipe(
-          map((enrollee: HttpEnrollee) => {
-            // Store the enrollee for access throughout creation and updating of a
-            // site, which will allows provide the most up-to-date site
-            // TODO replace each enrollee GET for the service
-            // this.paperEnrolmentService.enrollee = enrollee;
-
-            return this.routeDestination(routePath, enrollee);
-          })
+          map((enrollee: HttpEnrollee) => this.routeDestination(routePath, enrollee, params))
         );
-    } else {
-      return new Promise(async (resolve, reject) => resolve(true));
     }
+    return of(true);
   }
 
   /**
    * @description
    * Determine the route destination based on the enrolment.
    */
-  private routeDestination(routePath: string, enrolment: HttpEnrollee) {
-    return !!enrolment;
+  private routeDestination(routePath: string, httpEnrollee: HttpEnrollee, params: Params) {
+    if (!httpEnrollee?.approvedDate) {
+      return true;
+    } else {
+      return this.navigate(`/${this.config.routes.paperEnrolment}/${+params.eid}/${PaperEnrolmentRoutes.NEXT_STEPS}`, `${routePath}`, params);
```

Comment:
Don't need to wrap `routePath` with ticks since it is already a string.  Similarly don't need to cast `eid` since it will be a string so you're converting it to a number then back to a string.